### PR TITLE
Added selection of IOV sequence snapshot

### PR DESCRIPTION
**Pull request by ggovi**
This PR introduce the functionality for selecting the IOV sequence ( Tag ) snapshot. It is enabled by specifying a snapshot validity in the Global Tag ( none for the moment ) or to set a user-defined parameter in the cfg file. The default behaviour is unchanged.

### CODE REVIEW

**Comment by ggovi** [comment]
+1

**Comment by mmusich** [comment]
@cmsbuild please test

**Comment by mmusich** [comment]
+1 
provided jenkins agrees

**Comment by mmusich** [comment]
please test

**Comment by ggovi** [comment]
@cmsbuild 
the error in testConditionDatabase_0 is due to the fact that the tests have been executed twice ( in the same folder ). 
Can the test be executed in a fresh area?

**Comment by ggovi** [comment]
@cmsbuild please test
changed test file names to avoid clashes while running in multi processes

**Comment by ggovi** [comment]
+1

**Comment by ggovi** [comment]
@mmusich @diguida
please act...

**Comment by mmusich** [comment]
@ggovi let's wait the bot succeeds this time

**Comment by ggovi** [comment]
@cmsbuild please test

**Comment by ggovi** [comment]
not sure if the previous test got stuck...?

**Comment by nclopezo** [comment]
Hi @ggovi, They are running now.

**Comment by mmusich** [comment]
+1

**Comment by davidlange6** [comment]
+1

File: CondCore/ESSources/python/PoolDBESSource_cfi.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import FWCore.ParameterSet.Config as cms
 
-from CondCore.DBCommon.CondDBSetup_cfi import *
+from CondCore.CondDB.CondDB_cfi import *
```
Comment (Martin-Grunewald) [inline CondCore/ESSources/python/PoolDBESSource_cfi.py]: CondCore.CondDB.CondDB_cfi does not contain a "CondDBSetup" PSet needed later in this cfi file.
It does include a CondDB PSet. Should the latter be use, or should the above change be reverted?
(75X/76X).

File: CondCore/ESSources/python/PoolDBESSource_cfi.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 
 GlobalTag = cms.ESSource("PoolDBESSource",
     CondDBSetup,
```
Comment (Martin-Grunewald) [inline CondCore/ESSources/python/PoolDBESSource_cfi.py]: ...namely here!

**Comment by ggovi** [comment]
@Martin-Grunewald 
If I'm not wrong, the search of references to CondCore.ESSources.PoolDBESSource_cfi in CMSSW does not show a single entry. Would be ok to make separate PR to fix this? It is anyhow for usage in cfg files that are (currently ) outside the release...

**Comment by Martin-Grunewald** [comment]
ConfDB parsing uses CondCore.ESSources.PoolDBESSource_cfi to get the parameters of the
GT PoolDBESSource module (IOW, we need at least a correct cfi file, or a fillDescriptions method).
Separate PRs are fine, of course (75X and 76X).

**Comment by Martin-Grunewald** [comment]
@ggovi 
Any progress on the fix for 75/76?

**Comment by ggovi** [comment]
@Martin-Grunewald 
i should be able to do it this afternoon, will be done by today!

**Pull request merged**